leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/request/LwM2mResponseBuilder.java

```diff
@@ -263,7 +249,31 @@ public void visit(ReadCompositeRequest request) {
 
     @Override
     public void visit(ObserveCompositeRequest request) {
-        throw new UnsupportedOperationException("Not implemented yet.");
+        if (coapResponse.isError()) {
+            // handle error response:
+            lwM2mresponse = new ObserveCompositeResponse(toLwM2mResponseCode(coapResponse.getCode()), null,
+                    coapResponse.getPayloadString(), coapResponse, null
+            );
+
+        } else if (isResponseCodeContent() || isResponseCodeChanged()) {
```

Comment:
I don't think we need to support response code CHANGED :
The spec says that the right code is CONTENT. (see http://www.openmobilealliance.org/release/LightweightM2M/V1_1_1-20190617-A/HTML-Version/OMA-TS-LightweightM2M_Transport-V1_1_1-20190617-A.html#Table-645-1-Operation-to-Method-and-URI-Mapping-Information-Reporting-Interface)

The reason why we still support CHANGED for "single" observe response is because of historical spec changes : 
See https://github.com/eclipse/leshan/blob/a81877c5dc635f381f4dd451edb8adcfa6fbcf88/leshan-core/src/main/java/org/eclipse/leshan/core/response/ObserveResponse.java#L46L50

(I don't know if there is other code about this, if yes it should be removed I think)